### PR TITLE
Bytes unicode conversion

### PR DESCRIPTION
**Pull request by kleesc**
### Description of Changes

* Makes the conversions from str to bytes when needed.

#### Changes:

* ..
* ..

#### Issue: <link to story or task>

**TESTING** ->

**BREAKING CHANGE** ->

---

## Reviewer Checklist

- [ ] It works!
- [ ] Comments provide sufficient explanations for the next contributor
- [ ] Tests cover changes and corner cases
- [ ] Follows Quay syntax patterns and format

### CODE REVIEW

File: auth/test/test_basic.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
 def _token(username, password):
     assert isinstance(username, str)
     assert isinstance(password, str)
-    return "basic " + b64encode("%s:%s" % (username, password))
+    token_bytes = b"%s:%s" % (username.encode('ascii'), password.encode('ascii'))
```
Comment (josephschorr) [inline auth/test/test_basic.py]: What happens if the username or password is in unicode? While we don't support unicode usernames, we should make sure we don't raise an exception. Can you add a test for that (if we don't have one)?

File: avatars/avatars.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def get_data(self, name, email_or_id, kind="user"):
         # Note: email_or_id may be None if gotten from external auth when email is disabled,
         # so use the username in that case.
         username_email_or_id = email_or_id or name
```
Comment (josephschorr) [inline avatars/avatars.py]: Ditto. Test needed for a unicode input here

File: digest/digest_tools.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def sha256_digest_from_generator(content_generator):
   """
     digest = hashlib.sha256()
```
Comment (josephschorr) [inline digest/digest_tools.py]: Please add a test for creating a digest with a unicode input string

File: endpoints/api/tag.py [hunk 4]
```diff
@@ -32,10 +32,10 @@ def _tag_dict(tag):
         "reversion": tag.reversion,
     }
 
-    if tag.lifetime_start_ts > 0:
+    if tag.lifetime_start_ts and tag.lifetime_start_ts > 0:
```
Comment (josephschorr) [inline endpoints/api/tag.py]: Huh! `None > 0` worked in Python 2? Crazy

File: endpoints/v2/v2auth.py [hunk 5]
```diff
@@ -133,7 +133,7 @@ def generate_registry_jwt(auth_result):
     token = generate_bearer_token(
         audience_param, subject, context, access, TOKEN_VALIDITY_LIFETIME_S, instance_keys
     )
-    return jsonify({"token": token})
+    return jsonify({"token": token.decode('ascii')})
```
Comment (josephschorr) [inline endpoints/v2/v2auth.py]: Are we 100% sure JWT tokens are ASCII compatible? We should check the spec and link to the relative section

Comment (kleesc) [reply endpoints/v2/v2auth.py]: The tokens should be a base64 encoded string joined by a `.` AFAIK.

Comment (alecmerdler) [reply endpoints/v2/v2auth.py]: The first line of the [abstract in the RFC](https://tools.ietf.org/html/rfc7519): 
> JSON Web Token (JWT) is a compact, **URL-safe** means...
Yeah it's just base 64 character set + periods.

**Comment by kurtismullins** [comment]
@kleesc Can you run the `black` command on your branch? I see this lint error in CI:
```
9 files would be reformatted, 827 files would be left unchanged.
```
The command that is used by CI is available in the `scripts/ci` file.

File: data/fields.py [hunk 6]
```diff
@@ -72,12 +72,12 @@ class Base64BinaryField(TextField):
     def db_value(self, value):
         if value is None:
             return None
-        return base64.b64encode(value)
+        return base64.b64encode(value).decode('ascii')
```
Comment (alecmerdler) [inline data/fields.py]: I ran into this in my conversion work as well. This means every time we want to use this field, we need to make sure we are expecting actual `bytes` when using this field. Does that make sense? Or should we just keep everything as strings?

To avoid headaches, this guy suggested using the ["Unicode Sandwich"](https://nedbatchelder.com/text/unipain.html#h_pain_relief) (decode incoming binary immediately -> Python code manipulates decoded strings -> encode outgoing strings it

Comment (kurtismullins) [reply data/fields.py]: One potential pitfall is that not all unicode can be accurately converted to ASCII characters. We should verify the behavior on a case-by-case basis. My hunch is that it's fine but it'd be worth verifying the data being base64 encoded.

Comment (josephschorr) [reply data/fields.py]: If the data is passed through `b64encode`, it should be fine, as that call will fail if it cannot encode the data
